Source/Core/Core/MemoryWatcher.cpp

```diff
@@ -0,0 +1,89 @@
+// Copyright 2015 Dolphin Emulator Project
+// Licensed under GPLv2+
+// Refer to the license.txt file included.
+
+#include <fstream>
+#include <iostream>
+
+#include "Common/FileUtil.h"
+#include "Core/MemoryWatcher.h"
+#include "Core/HW/Memmap.h"
+
+// We don't want to kill the cpu, so sleep for this long after polling.
+static const int SLEEP_DURATION = 10; // ms
+
+MemoryWatcher::MemoryWatcher()
+{
+	if (!LoadAddresses(File::GetUserPath(F_MEMORYWATCHERLOCATIONS_IDX)))
+		return;
+	if (!OpenSocket(File::GetUserPath(F_MEMORYWATCHERSOCKET_IDX)))
+		return;
+	m_watcher_thread = std::thread(std::bind(&MemoryWatcher::WatcherThread, this));
+}
+
+MemoryWatcher::~MemoryWatcher()
+{
+	if (!m_running)
```

Comment:
This is a race condition. m_running will be set within the thread, not within the constructor.